storage/adapters/pgp_adapter.js

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const path = require('path');
+const Promise = require('bluebird');
+const monitor = require('pg-monitor');
+const pgpLib = require('pg-promise');
+const QueryFile = require('pg-promise').QueryFile;
+const BaseAdapter = require('./base_adapter');
+
+const resultCountToMethodMap = {
+	0: 'none',
+	1: 'one',
+	3: 'any',
+	[undefined]: 'any',
+};
+
+class PgpAdapter extends BaseAdapter {
+	/**
+	 *
+	 * @param {Object} options
+	 * @param {Boolean} options.inTest
+	 * @param {string} options.logger
+	 * @param {string} options.sqlDirectory
+	 */
+	constructor(options) {
+		super({
+			engineName: 'pgp',
+			inTest: options.inTest,
+		});
+
+		this.options = options;
+		this.logger = options.logger;
+		this.sqlDirectory = options.sqlDirectory;
+
+		this.pgpOptions = {
+			capSQL: true,
+			promiseLib: Promise,
+			noLocking: options.inTest,
+			connect: () => {
+				this.emit(this.EVENT_CONNECT);
+			},
+			error: () => {
+				this.emit(this.EVENT_ERROR);
+			},
+			disconnect: () => {
+				this.emit(this.EVENT_DISCONNECT);
+			},
+		};
+
+		this.pgp = pgpLib(this.pgpOptions);
+		this.db = null;
```

Comment:
Consistent usage of `null` vs `undefined`.
After we move to TypeScript, we started to use this rule
https://palantir.github.io/tslint/rules/no-null-keyword/